examples/examples-release-18/src/test/java/io/kubernetes/client/examples/ExampleTest.java

```diff
@@ -14,33 +14,36 @@
 
 import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
 import static com.github.tomakehurst.wiremock.client.WireMock.get;
-import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
 import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
-import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 
-import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import com.github.tomakehurst.wiremock.junit5.WireMockExtension;
 import io.kubernetes.client.openapi.ApiClient;
 import io.kubernetes.client.openapi.ApiException;
 import io.kubernetes.client.openapi.Configuration;
 import io.kubernetes.client.openapi.apis.CoreV1Api;
 import io.kubernetes.client.openapi.models.V1Namespace;
 import io.kubernetes.client.openapi.models.V1ObjectMeta;
-import org.junit.Rule;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
-public class ExampleTest {
+class ExampleTest {
   private static final int PORT = 8089;
-  @Rule public WireMockRule wireMockRule = new WireMockRule(PORT);
+
+  @RegisterExtension
+  static WireMockExtension apiServer =
+      WireMockExtension.newInstance().options(WireMockConfiguration.options().port(PORT)).build();
 
   @Test
-  public void exactUrlOnly() throws ApiException {
+  void exactUrlOnly() throws ApiException {
     ApiClient client = new ApiClient();
     client.setBasePath("http://localhost:" + PORT);
     Configuration.setDefaultApiClient(client);
 
     V1Namespace ns1 = new V1Namespace().metadata(new V1ObjectMeta().name("name"));
 
-    stubFor(
+    apiServer.stubFor(
```

Comment:
this is an example which bit me.. basically when using the junit rule, you can mix/match using static method or the method on the rule. in the extension you must use the method and not use the static.

Reply:
when you don't do this, things will fail about reaching port 8080 and you have to think through why. I think this maybe should become an openrewrite recipe for wiremock (removing the static method footgun and always even in junit 4 using the instance to do stubbing and verification cc @timtebeek)

Reply:
Thanks for the suggestion! That's bit me before in the past as well; would be neat to have a recipe that changes any static method use to use through `rule` / `extension` ahead of any other WireMock migration steps. A good addition to this earlier issue.
- https://github.com/openrewrite/rewrite-testing-frameworks/issues/278